app/controllers/guests_controller.rb

```diff
@@ -55,13 +55,9 @@ def guest_in_query(g, query)
   end
 
   def correct_guest
-    meta = current_user.try(:meta)
-    id = params[:id].to_i
+    return if current_user && (current_user.admin? || current_user.sub_admin?)
 
-    return if (current_user && (current_user.admin? || current_user.sub_admin?)) || (meta && meta.is_a?(Manager))
-
-    
-    redirect_to user_session_path if meta.nil? || (meta.is_a?(Guest) && meta.id != id)
-    redirect_to user_session_path if meta.is_a?(Manager) && !meta.hosts.pluck(:id).include?(id)
+    redirect_to user_session_path unless current_user.guests.any? {|guest| guest.id == params[:id].to_i}
+    # redirect_to user_session_path if meta.is_a?(Manager) && !meta.hosts.pluck(:id).include?(id)
```

Comment:
because there are now several guests for each user, make sure you can see your own.
question: why take out manager? (we need manager's to have access to everyone)
@talharamati